blockchain/blockchain.go

```diff
@@ -212,3 +228,120 @@ func (b *Blockchain) SanityCheckNewHeight(block *core.Block, stateUpdate *core.S
+		return err
+	}
+
+	if txnBytes, err := encoder.Marshal(transaction); err != nil {
+		return err
+	} else if err = txn.Set(db.Transactions.Key(bnIndexBytes), txnBytes); err != nil {
+		return err
+	}
+	return nil
+}
+
+// getBlockNumberAndIndex gets the block number and index for a given transaction hash
+func getBlockNumberAndIndex(txn db.Transaction, hash *felt.Felt) ([]byte, error) {
+	bnIndexBytes, err := txn.Get(db.TransactionsIndexByHash.Key(hash.Marshal()))
+	if err != nil {
+		return nil, err
+	}
+
+	return bnIndexBytes, nil
+}
+
+// getTransactionByBnIndexBytes gets the transaction for a given block number and index.
+func getTransactionByBnIndexBytes(txn db.Transaction, bnIndexBytes []byte) (core.Transaction, error) {
+	if value, err := txn.Get(db.Transactions.Key(bnIndexBytes)); err != nil {
+		return nil, err
+	} else {
+		var transaction core.Transaction
+		if err := encoder.Unmarshal(value, &transaction); err != nil {
+			return nil, err
+		}
+		return transaction, nil
+	}
+}
+
+// getTransactionByHash gets the transaction for a given hash.
+func getTransactionByHash(txn db.Transaction, hash *felt.Felt) (core.Transaction, error) {
+	bnIndexBytes, err := getBlockNumberAndIndex(txn, hash)
+	if err != nil {
+		return nil, err
+	}
+	return getTransactionByBnIndexBytes(txn, bnIndexBytes)
+}
+
+// GetReceipt gets the transaction receipt for a given transaction hash.
+func (b *Blockchain) GetReceipt(hash *felt.Felt) (receipt *core.TransactionReceipt, err error) {
+	return receipt, b.database.View(func(txn db.Transaction) error {
+		receipt, err = getReceiptByHash(txn, hash)
+		return err
+	})
+}
+
+// storeReceipt stores the given transaction receipt in the database.
+func storeReceipt(txn db.Transaction, blockNumber, index uint64, receipt core.TransactionReceipt) error {
+	bnIndexBytes := joinBlockNumberAndIndex(blockNumber, index)
+	if err := txn.Set(db.TransactionsIndexByHash.Key((receipt.TransactionHash).Marshal()), bnIndexBytes); err != nil {
+		return err
+	}
+
+	if txnBytes, err := encoder.Marshal(receipt); err != nil {
+		return err
+	} else {
+		return txn.Set(db.TransactionReceipts.Key(bnIndexBytes), txnBytes)
+	}
+}
+
+// getReceipt gets the transaction receipt for a given block number and index.
+func getReceipt(txn db.Transaction, bnIndexBytes []byte) (*core.TransactionReceipt, error) {
+	value, err := txn.Get(db.TransactionReceipts.Key(bnIndexBytes))
+	if err != nil {
+		return nil, err
+	}
+
+	var receipt *core.TransactionReceipt
+	if err := encoder.Unmarshal(value, &receipt); err != nil {
+		return nil, err
+	}
+	return receipt, nil
+}
+
+// getReceiptByHash gets the transaction receipt for a given hash.
+func getReceiptByHash(txn db.Transaction, hash *felt.Felt) (*core.TransactionReceipt, error) {
+	if bnIndexBytes, err := getBlockNumberAndIndex(txn, hash); err != nil {
+		return nil, err
+	} else {
+		return getReceipt(txn, bnIndexBytes)
+	}
+}
+
+func joinBlockNumberAndIndex(blockNumber, index uint64) []byte {
```

Comment:
let's make this a type like;

type TxKey struct {
   blockNum uint64
   index uint64
}

and use Marshal & Unmarshal, so that we can easily recover blockNum and indexes when iterating over keys just by unmarshaling a TxKey